mtgjson5/providers/tcgplayer.py

```diff
@@ -171,6 +240,61 @@ def generate_tcgplayer_to_mtgjson_map(
     return dump_map
 
 
+def get_tcgplayer_buylist_prices_map(
+    group_id_and_name: Tuple[str, str]
+) -> Dict[str, MtgjsonPricesObject]:
+    """
+    takes a group id and name and finds all buylist data for that group
+    :param group_id_and_name: TCGPlayer Set ID & Name to build
+    :return: returns a map of tcgplayer buylist data to card uuids
+    """
+    LOGGER.info(f"Building {group_id_and_name[1]} TCGPlayer Buylist data")
+    api_response = TCGPlayerProvider().download(
+        f"https://api.tcgplayer.com/pricing/buy/group/{group_id_and_name[0]}"
+    )
+    if not api_response:
+        return {}
+    response = json.loads(api_response)
+    if not response["results"]:
+        return {}
+    prices_map: Dict[str, MtgjsonPricesObject] = {}
+    # when building the full system, only build this once and take it as an arg/write to a file
+    uuid_map = generate_tcgplayer_to_mtgjson_map(
+        OUTPUT_PATH.joinpath("AllPrintings.json")
+    )
+    # this is set specific, built each time unlike the uuid map
+    sku_map = generate_tcgplayer_sku_map(get_tcgplayer_sku_data(group_id_and_name))
```

Comment:
Function to a function like this is a bit yucky. I'd try to clarify what's going on here